### PR TITLE
[ᚬmaster] Rc/v0.24.2

### PR DESCRIPTION
**Pull request by duanyytop**
# [v0.24.2](https://github.com/nervosnetwork/ckb-sdk-java/compare/v0.24.1...v0.24.2) (2019-11-07)

### Feature

* Add AddressGenerator and AddressParser([dbf4fe5](https://github.com/nervosnetwork/ckb-sdk-java/commit/dbf4fe57cde7965387e1fa2116b9d9a1eae9a30c))
* Update address generator and parser for example([7cdf948](https://github.com/nervosnetwork/ckb-sdk-java/commit/7cdf9488c7b3618f89c6c03492d96949fce85e8e))

### CODE REVIEW

File: ckb/src/main/java/org/nervos/ckb/utils/address/AddressParser.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import org.nervos.ckb.address.Network;
+import org.nervos.ckb.exceptions.AddressFormatException;
+import org.nervos.ckb.type.Script;
+import org.nervos.ckb.utils.Bech32;
+import org.nervos.ckb.utils.Numeric;
+
+/** Copyright © 2019 Nervos Foundation. All rights reserved. */
+public class AddressParser extends AddressBaseOperator {
+
+  private static String parsePrefix(String address) {
```
Comment (shaojunda) [inline ckb/src/main/java/org/nervos/ckb/utils/address/AddressParser.java]: `parsePrefix` I thought that the parsing is prefix(hrp)

File: ckb/src/main/java/org/nervos/ckb/utils/address/AddressParser.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    byte[] data = convertBits(com.google.common.primitives.Bytes.asList(parsed.data), 5, 8, false);
+    if (data.length == 0) {
+      return null;
+    }
+    Bech32.Bech32Data bech32Data = new Bech32.Bech32Data(parsed.hrp, data);
+    return Numeric.toHexStringNoPrefix(bech32Data.data);
+  }
+
+  public static AddressParseResult parse(String address) throws AddressFormatException {
+    String payload = parsePrefix(address);
```
Comment (shaojunda) [inline ckb/src/main/java/org/nervos/ckb/utils/address/AddressParser.java]: `parsePrefix(address)` return `payload` make me a bit confused

File: ckb/src/main/java/org/nervos/ckb/utils/address/AddressParser.java (hunk 1 above)
Comment (duanyytop) [reply ckb/src/main/java/org/nervos/ckb/utils/address/AddressParser.java]: Updated. https://github.com/nervosnetwork/ckb-sdk-java/pull/216/commits/eebfa90461288c1274d1fc8153d928d0db722285

File: ckb/src/main/java/org/nervos/ckb/utils/address/AddressParser.java (hunk 2 above)
Comment (duanyytop) [reply ckb/src/main/java/org/nervos/ckb/utils/address/AddressParser.java]: Updated. https://github.com/nervosnetwork/ckb-sdk-java/pull/216/commits/eebfa90461288c1274d1fc8153d928d0db722285

File: ckb/src/main/java/org/nervos/ckb/utils/address/AddressParser.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    String payload = parsePayload(address);
+    if (payload == null) {
+      throw new AddressFormatException("Address bech32 decode fail");
+    }
+    String type = payload.substring(0, 2);
+    if (TYPE_SHORT.equals(type)) {
+      String codeHashIndex = payload.substring(2, 4);
+      String args = Numeric.prependHexPrefix(payload.substring(4));
+      if (Numeric.cleanHexPrefix(args).length() != 40) {
+        throw new AddressFormatException("Short address args byte length must be equal to 20");
```
Comment (shaojunda) [inline ckb/src/main/java/org/nervos/ckb/utils/address/AddressParser.java]: if statement check hex string length but error message show args legal length, would it be better to unify it here?

**Pull request merged**